modelscope/msdatasets/dataset_cls/custom_datasets/audio/asr_dataset.py

```diff
@@ -33,16 +34,20 @@ def load(cls,
              dataset_name,
              namespace='speech_asr',
              train_set='train',
-             dev_set='validation'):
+             dev_set='validation',
+            download_mode: Optional[DownloadMode] = None):
         if os.path.exists(dataset_name):
-            data_dir = dataset_name
-            ds_dict = {}
-            ds_dict['train'] = cls.load_core(data_dir, train_set)
-            ds_dict['validation'] = cls.load_core(data_dir, dev_set)
-            ds_dict['raw_data_dir'] = data_dir
-            return ds_dict
+            if download_mode != DownloadMode.FORCE_REDOWNLOAD:
```

Comment:
line 40 ~ 52, seems like the old version is proper, and you can pass the download_mode param to MsDataset.load() function. More specifically,  code "if os.path.exists(dataset_name):" which means to download user-define dataset from local disk, just ignore it.